client/pages/favorite.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import Carousel from '../components/carousel';
+import FavoriteDisplay from '../components/favorite-display';
+import AppContext from '../lib/app-context';
+import Redirect from '../components/redirect';
+
+export default class FavoriteSearch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchResult: null,
+      searchInput: '',
+      assistInput: '',
+      searchType: 'Artist'
+
+    };
+    this.searchTypeChoose = this.searchTypeChoose.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  searchTypeChoose(event) {
+    this.setState({ searchType: event.target.getAttribute('data-type') });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const input = this.state.searchInput;
+    const assistInput = this.state.assistInput;
+    let fetchPath;
+    let fileName;
+    if (this.state.searchType === 'Album') {
+      fetchPath = `https://theaudiodb.p.rapidapi.com/searchalbum.php?s=${assistInput}&a=${input}`;
+      fileName = 'album';
+
+    } else {
+      fetchPath = `https://theaudiodb.p.rapidapi.com/search.php?s=${input}`;
+      fileName = 'artists';
+    }
+
+    fetch(fetchPath, {
+      method: 'GET',
+      headers: {
+        'x-rapidapi-host': 'theaudiodb.p.rapidapi.com',
+        'x-rapidapi-key': '3ac221b76dmsh761303974d244fbp12a694jsna1c5bc30b528'
+      }
```

Comment:
I am afraid that your key is now leaked. It's very important to keep api keys private if you can. Usually it is better to make a custom route in your server code that gets the data for you. You can use the `node-fetch` package to use `fetch()` in the server code to talk to the audiodb api but still keep your key secret